File: BEmu_cpp/src/BloombergTypes/Request.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/BloombergTypes/Request.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "BloombergTypes/Request.h"

namespace BEmu
{
	Request::Request()
	{
	}

	Request::Request(boost::shared_ptr<RequestPtr> ptr) : _shptr(ptr)
	{
	}

	Request::Request(const Request& arg) : _shptr(arg._shptr)
	{
	}

	Request& Request::operator=(const Request &rhs)
	{
		if (this != &rhs)
		{
			this->_shptr = boost::shared_ptr<RequestPtr>(rhs._shptr);
		}
		return *this;
	}

	Request::~Request()
	{
	}
	
	boost::shared_ptr<RequestPtr> Request::getRequestPtr() const
	{
		return this->_shptr;
	}

	void Request::append(const char* name, const char* value)
	{
		this->_shptr->append(name, value);
	}

	void Request::append(const Name& name, const char* value)
	{
		this->_shptr->append(name, value);
	}


	void Request::set(const char* name, const char* value)
	{
		this->_shptr->set(name, value);
	}

	void Request::set(const Name& name, const char* value)
	{
		this->_shptr->set(name, value);
	}


	void Request::set(const char* name, const Datetime& value)
	{
		this->_shptr->set(name, value);
	}
	void Request::set(const Name& name, const Datetime& value)
	{
		this->_shptr->set(name, value);
	}


	void Request::set(const char* name, bool value)
	{
		this->_shptr->set(name, value);
	}

	void Request::set(const Name& name, bool value)
	{
		this->_shptr->set(name, value);
	}


	void Request::set(const char* name, int value)
	{
		this->_shptr->set(name, value);
	}

	void Request::set(const Name& name, int value)
	{
		this->_shptr->set(name, value);
	}


	Element Request::getElement(const char* name)
	{
		return this->_shptr->getElement(name);
	}

	Element Request::getElement(const Name& name)
	{
		return this->_shptr->getElement(name);
	}

	std::ostream& Request::print(std::ostream& stream, int level, int spacesPerLevel) const
	{
		this->_shptr->print(stream, level, spacesPerLevel);
		return stream;
	}

	std::ostream& operator<<(std::ostream& stream, const Request &request)
	{
		request.print(stream);
		return stream;
	}

}

File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/Request.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/Request.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.CorrelationID;
import com.bloomberglp.blpapi.Element;

public abstract class Request 
{
	protected Request()
	{
		this._operation = new Operation();
	}
	
	private CorrelationID _correlationId;
	private final Operation _operation;
	
	public Operation operation()
	{
		return this._operation;
	}
	
	public Element asElement() throws Exception
	{
		throw new Exception("not implemented");
	}
	
	public CorrelationID correlationId()
	{
		return this._correlationId;
	}
	
	void correlationId(CorrelationID corr)
	{
		this._correlationId = corr;
	}

    public void append(String name, String elementValue) throws Exception
    {
        throw new Exception("BEmu.Request.Append: Append is not implemented");
    }
    
    public void set(String name, String elementValue) throws Exception
    {
        throw new Exception("BEmu.Request.Set: string is not implemented");
    }
    
    public void set(String name, boolean elementValue) throws Exception
    {
        throw new Exception("BEmu.Request.Set: bool is not implemented");
    }

    public void set(String name, int elementValue) throws Exception
    {
        throw new Exception("BEmu.Request.Set: int is not implemented");
    }

    public void set(String name, Datetime elementValue) throws Exception
    {
        throw new Exception("BEmu.Request.Set: Datetime is not implemented");
    }

    public void set(Name name, String elementValue) throws Exception
    {
        this.set(name.toString(), elementValue);
    }

    public void set(Name name, boolean elementValue) throws Exception
    {
        this.set(name.toString(), elementValue);
    }

    public void set(Name name, int elementValue) throws Exception
    {
        this.set(name.toString(), elementValue);
    }

    public void set(Name name, Datetime elementValue) throws Exception
    {
        this.set(name.toString(), elementValue);
    }

    public Element getElement(Name name) throws Exception
    {
        return this.getElement(name.toString());
    }

    public Element getElement(String name) throws Exception
    {
    	throw new Exception("BEmu.Request.this[]: by string is not implemented");
    }

    public boolean hasElement(String name) throws Exception
    {
        throw new Exception("BEmu.Request.HasElement: string is not implemented");
    }
}
